### DISCUSSION TITLE
CSLA 5.5.3 and Blazor Server

### DISCUSSION THREAD

**Discussion by oskarjf** [MarimerLLC/csla#3587] [Questions]
Hi

I've a Blazor Server where I'm injecting a service into a Razor component.  This service inherits from CommandBase and has a injected service for the method DataPortal_Execute.  For some reason this injected service as a parameter is not being resolved by CSLA.
Any ideas as to where I should be looking?

**Comment by oskarjf** [comment]
Think the issue is just related to this again:
https://github.com/MarimerLLC/csla/discussions/3517

**Reply by rockfordlhotka** [reply]
I _strongly_ recommend moving to CSLA 6 when using Blazor server. There were a number of really big issues with CSLA 5 and Blazor server, and the only way to solve them was through the breaking changes made in CSLA 6.

Well, really you should move to CSLA 7, because that includes a bunch of bug fixes that also make working with Blazor easier.

**Reply by oskarjf** [reply]
Good to know. I'll look into how much an effort it is to go to v6 vs v7

**Reply by oskarjf** [reply]
@rockfordlhotka Would you recommend upgrading to v6 first from 5.5.3 or go directly to v7?

**Reply by rockfordlhotka** [reply]
I think I'd go directly to CSLA 7.

Versions 6 and 7 are _mostly_ the same - it was CSLA 6 that _requires_ the use of DI (to solve the Blazor server interactive issues). There's no value I can think of to go to 6 and then 7, vs just going to 7 directly.
